tests/integration.rs

```diff
@@ -736,9 +736,20 @@ mod tests {
             );
         }
 
-        fn reboot_linux(&self, current_reboot_count: u32, custom_timeout: Option<i32>) {
+        fn reboot_linux(
+            &self,
+            current_reboot_count: u32,
+            custom_timeout: Option<i32>,
+            matching_pattern: Option<&str>,
+        ) {
+            let cmd = if let Some(pattern) = matching_pattern {
+                format!("sudo journalctl | grep -c -- \"{}\"", pattern)
```

Comment:
Can you replace this with a grep that checks both patterns as a regexp rather than making it take a parameter,

Reply:
Done!

Reply:
Hrm so this is not working properly since Focal has both patterns :(
So I decided to move to using `systemd-reboot.service: Succeeded` pattern only, but unfortunately `test_reboot` tests both Bionic and Focal, and it appears Bionic does not have `systemd-reboot.service` log.
Either I go back to the original code introducing a new parameter, or I can remove Bionic from `test_reboot`. WDYT?

Reply:
What about about `journalctl --list-boots` ?

Reply:
I updated with this. PR should be ready to be merged now.